File: projects/applications/ASCII-String-UI-Editor/utils/presets.py
"""Character set preset definitions"""

SINGLE_LINE = {
    "top_left": "┌",
    "top_right": "┐",
    "bottom_left": "└",
    "bottom_right": "┘",
    "horizontal": "─",
    "vertical": "│",
    "tjunction_up": "┴",
    "tjunction_down": "┬",
    "tjunction_left": "┤",
    "tjunction_right": "├",
    "cross": "┼",
}

DOUBLE_LINE = {
    "top_left": "╔",
    "top_right": "╗",
    "bottom_left": "╚",
    "bottom_right": "╝",
    "horizontal": "═",
    "vertical": "║",
    "tjunction_up": "╩",
    "tjunction_down": "╦",
    "tjunction_left": "╣",
    "tjunction_right": "╠",
    "cross": "╬",
}

HEAVY_LINE = {
    "top_left": "┏",
    "top_right": "┓",
    "bottom_left": "┗",
    "bottom_right": "┛",
    "horizontal": "━",
    "vertical": "┃",
    "tjunction_up": "┻",
    "tjunction_down": "┳",
    "tjunction_left": "┫",
    "tjunction_right": "┣",
    "cross": "╋",
}

ROUNDED_LINE = {
    "top_left": "╭",
    "top_right": "╮",
    "bottom_left": "╰",
    "bottom_right": "╯",
    "horizontal": "─",
    "vertical": "│",
    "tjunction_up": "┴",
    "tjunction_down": "┬",
    "tjunction_left": "┤",
    "tjunction_right": "├",
    "cross": "┼",
}

MIXED_DOUBLE_SINGLE = {
    "top_left": "╒",
    "top_right": "╕",
    "bottom_left": "╘",
    "bottom_right": "╛",
    "horizontal": "═",
    "vertical": "│",
    "tjunction_up": "╧",
    "tjunction_down": "╤",
    "tjunction_left": "╡",
    "tjunction_right": "╞",
    "cross": "╪",
}

ASCII_BASIC = {
    "top_left": "+",
    "top_right": "+",
    "bottom_left": "+",
    "bottom_right": "+",
    "horizontal": "-",
    "vertical": "|",
    "tjunction_up": "+",
    "tjunction_down": "+",
    "tjunction_left": "+",
    "tjunction_right": "+",
    "cross": "+",
}

PRESETS = {
    "SINGLE_LINE": SINGLE_LINE,
    "DOUBLE_LINE": DOUBLE_LINE,
    "HEAVY_LINE": HEAVY_LINE,
    "ROUNDED_LINE": ROUNDED_LINE,
    "MIXED_DOUBLE_SINGLE": MIXED_DOUBLE_SINGLE,
    "ASCII_BASIC": ASCII_BASIC,
}

def format_charset_python(charset: dict) -> str:
    """Format charset as Python dict string"""
    lines = ["{"]
    for key, value in charset.items():
        lines.append(f'    "{key}": "{value}",')
    lines.append("}")
    return "\n".join(lines)

def format_charset_javascript(charset: dict) -> str:
    """Format charset as JavaScript object string"""
    lines = ["{"]
    for key, value in charset.items():
        lines.append(f'    "{key}": "{value}",')
    lines.append("}")
    return "\n".join(lines)

File: projects/applications/ASCII-String-UI-Editor/src/presets.ts
export interface CharacterSet {
  top_left: string;
  top_right: string;
  bottom_left: string;
  bottom_right: string;
  horizontal: string;
  vertical: string;
  tjunction_up: string;
  tjunction_down: string;
  tjunction_left: string;
  tjunction_right: string;
  cross: string;
}

export const SINGLE_LINE: CharacterSet = {
  top_left: '┌',
  top_right: '┐',
  bottom_left: '└',
  bottom_right: '┘',
  horizontal: '─',
  vertical: '│',
  tjunction_up: '┴',
  tjunction_down: '┬',
  tjunction_left: '┤',
  tjunction_right: '├',
  cross: '┼',
};

export const DOUBLE_LINE: CharacterSet = {
  top_left: '╔',
  top_right: '╗',
  bottom_left: '╚',
  bottom_right: '╝',
  horizontal: '═',
  vertical: '║',
  tjunction_up: '╩',
  tjunction_down: '╦',
  tjunction_left: '╣',
  tjunction_right: '╠',
  cross: '╬',
};

export const HEAVY_LINE: CharacterSet = {
  top_left: '┏',
  top_right: '┓',
  bottom_left: '┗',
  bottom_right: '┛',
  horizontal: '━',
  vertical: '┃',
  tjunction_up: '┻',
  tjunction_down: '┳',
  tjunction_left: '┫',
  tjunction_right: '┣',
  cross: '╋',
};

export const ROUNDED_LINE: CharacterSet = {
  top_left: '╭',
  top_right: '╮',
  bottom_left: '╰',
  bottom_right: '╯',
  horizontal: '─',
  vertical: '│',
  tjunction_up: '┴',
  tjunction_down: '┬',
  tjunction_left: '┤',
  tjunction_right: '├',
  cross: '┼',
};

export const MIXED_DOUBLE_SINGLE: CharacterSet = {
  top_left: '╒',
  top_right: '╕',
  bottom_left: '╘',
  bottom_right: '╛',
  horizontal: '═',
  vertical: '│',
  tjunction_up: '╧',
  tjunction_down: '╤',
  tjunction_left: '╡',
  tjunction_right: '╞',
  cross: '╪',
};

export const ASCII_BASIC: CharacterSet = {
  top_left: '+',
  top_right: '+',
  bottom_left: '+',
  bottom_right: '+',
  horizontal: '-',
  vertical: '|',
  tjunction_up: '+',
  tjunction_down: '+',
  tjunction_left: '+',
  tjunction_right: '+',
  cross: '+',
};

export const PRESETS: Record<string, { name: string; charset: CharacterSet }> = {
  SINGLE_LINE: { name: 'Single Line', charset: SINGLE_LINE },
  DOUBLE_LINE: { name: 'Double Line', charset: DOUBLE_LINE },
  HEAVY_LINE: { name: 'Heavy Line', charset: HEAVY_LINE },
  ROUNDED_LINE: { name: 'Rounded Corners', charset: ROUNDED_LINE },
  MIXED_DOUBLE_SINGLE: { name: 'Mixed Double/Single', charset: MIXED_DOUBLE_SINGLE },
  ASCII_BASIC: { name: 'ASCII Basic (+/-/|)', charset: ASCII_BASIC },
};
